### PR TITLE
Added Usage System (Bommels)

### PR DESCRIPTION
**Pull request by Bommels05**
Added A Usage System and Match Command

### CODE REVIEW

**Comment by Bommels05** [comment]
[ingwer.zip](https://github.com/TureBentzin/Ingwer/files/9639470/ingwer.zip)

File: src/main/java/de/bentzin/ingwer/command/node/preset/UsageNodeExecutor.java [hunk 1]
```diff
@@ -28,20 +35,49 @@ public UsageNodeExecutor(IngwerMessage usage) {
...
+                messages.add(builder.build());
+            }
+            else {
+                MessageBuilder builder = MessageBuilder.empty();
+                int i = entry.getKey();
+                while (i > 2) {
+                    builder.add(C.C, "  ");
+                    i--;
+                }
+                builder.add(C.C, "|").add(C.C, "-");
```
Comment (TureBentzin) [inline src/main/java/de/bentzin/ingwer/command/node/preset/UsageNodeExecutor.java:56]:
```suggestion
                builder.add(C.C, "|-");
```

File: src/main/java/de/bentzin/ingwer/command/node/preset/UsageNodeExecutor.java [hunk 2]
```diff
@@ -28,20 +35,49 @@ public UsageNodeExecutor(IngwerMessage usage) {
...
+                addContent(entry.getValue(), builder);
+                messages.add(builder.build());
+            }
+        }
+        FramedMessage message = new FramedMessage(messages);
+        return new UsageNodeExecutor(message);
+    }
+
+    private static List<Map.Entry<Integer, Node<?>>> getTree(int depth, Node<?> node) {
+        System.out.println(node);
```
Comment (TureBentzin) [inline src/main/java/de/bentzin/ingwer/command/node/preset/UsageNodeExecutor.java:66]: debug

File: src/main/java/de/bentzin/ingwer/command/node/preset/UsageNodeExecutor.java [hunk 3]
```diff
@@ -28,20 +35,49 @@ public UsageNodeExecutor(IngwerMessage usage) {
...
+        }
+        return nodes;
+    }
+
+    private static void addContent(Node<?> node, MessageBuilder builder) {
+        StringJoiner joiner = new StringJoiner(IngwerMessage.COLOR_MM + ", " + IngwerMessage.COLOR_MM_C);
+        for (String s : node.values()) {
+            joiner.add(s);
+        }
+        builder.add(C.C, "[").add(C.A, node.getName()).add(C.C, "]").add(C.A, "<dark_gray> » </dark_gray>" + joiner);
```
Comment (TureBentzin) [inline src/main/java/de/bentzin/ingwer/command/node/preset/UsageNodeExecutor.java:80]: may just use a regular add before the accent message

File: src/main/java/de/bentzin/ingwer/features/integrated/vault/VaultFeature.java [hunk 4]
```diff
@@ -72,6 +72,9 @@ private boolean setupChat() {
             return false;
         }
         RegisteredServiceProvider<Chat> rsp = Ingwer.javaPlugin.getServer().getServicesManager().getRegistration(Chat.class);
+        if (rsp == null) {
+            return false;
```
Comment (TureBentzin) [inline src/main/java/de/bentzin/ingwer/features/integrated/vault/VaultFeature.java:76]: good fix

**Pull request merged**